File: app/src/androidTest/java/com/pavelnazimok/uitesting/java/robots/FirstScreenRobot.java
package com.pavelnazimok.uitesting.java.robots;

import com.pavelnazimok.uitesting.R;

import static com.pavelnazimok.uitesting.java.utils.EspressoExtensions.getText;
import static com.pavelnazimok.uitesting.java.utils.EspressoExtensions.waitForView;
import static com.schibsted.spain.barista.assertion.BaristaVisibilityAssertions.assertDisplayed;
import static com.schibsted.spain.barista.interaction.BaristaClickInteractions.clickOn;

@SuppressWarnings("UnusedReturnValue")
public class FirstScreenRobot {

    public int getCoffeePrice() {
        return Integer.valueOf(getText(R.id.label_coffee_price));
    }

    public FirstScreenRobot clickCoffeeIncrement() {
        clickOn(R.id.button_coffee_increment);
        return this;
    }

    public FirstScreenRobot clickCoffeeDecrement() {
        clickOn(R.id.button_coffee_decrement);
        return this;
    }

    public FirstScreenRobot assertFirstScreenIsDisplayed() {
        waitForView(R.id.screen_first, 5000);
        return this;
    }

    public FirstScreenRobot assertCoffeeCountEquals(final int count) {
        assertDisplayed(R.id.label_coffee_count, String.valueOf(count));
        return this;
    }

    public FirstScreenRobot assertCoffeePriceEquals(final int price) {
        assertDisplayed(R.id.label_coffee_price, String.valueOf(price));
        return this;
    }

    public FirstScreenRobot assertTotalPriceEquals(final int price) {
        assertDisplayed(R.id.label_total_price, String.valueOf(price));
        return this;
    }
}

File: app/src/androidTest/java/com/pavelnazimok/uitesting/kotlin/robots/FirstScreenRobot.kt
package com.pavelnazimok.uitesting.kotlin.robots

import com.pavelnazimok.uitesting.R
import com.pavelnazimok.uitesting.kotlin.utils.getText
import com.pavelnazimok.uitesting.kotlin.utils.waitForView
import com.schibsted.spain.barista.assertion.BaristaVisibilityAssertions.assertDisplayed
import com.schibsted.spain.barista.interaction.BaristaClickInteractions.clickOn

fun onFirstScreen(func: FirstScreenRobot.() -> Unit): FirstScreenRobot {
    return FirstScreenRobot().apply {
        assertFirstScreenIsDisplayed()
        func()
    }
}

class FirstScreenRobot {

    fun getCoffeePrice(): Int {
        return Integer.valueOf(getText(R.id.label_coffee_price))
    }

    fun clickCoffeeIncrement() {
        clickOn(R.id.button_coffee_increment)
    }

    fun clickCoffeeDecrement() {
        clickOn(R.id.button_coffee_decrement)
    }

    fun assertFirstScreenIsDisplayed() {
        waitForView(R.id.screen_first)
    }

    fun assertCoffeeCountEquals(count: Int) {
        assertDisplayed(R.id.label_coffee_count, count.toString())
    }

    fun assertCoffeePriceEquals(price: Int) {
        assertDisplayed(R.id.label_coffee_price, price.toString())
    }

    fun assertTotalPriceEquals(price: Int) {
        assertDisplayed(R.id.label_total_price, price.toString())
    }
}
